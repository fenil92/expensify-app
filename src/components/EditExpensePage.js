import React from 'react';

const EditExpensePage = (props) => (
    <p>edit expense here {props.match.params.id}</p>
);


export default EditExpensePage;