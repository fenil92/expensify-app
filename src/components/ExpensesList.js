import React from 'react';
import { connect } from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
import ExpenseListItem from './ExpenseListItem';

export const ExpenseListComponent = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (<p>No Expenses</p>) : 
            (props.expenses.map((expense) => {
                return <ExpenseListItem key={expense.id} {...expense}></ExpenseListItem>
            }))
        }
       
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses : getVisibleExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseListComponent);