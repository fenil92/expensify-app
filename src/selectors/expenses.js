import moment from 'moment';
// unix timestamps (milliseconds)
// January 1st, 1970
// get visible expenses
const getVisibleExpenses = (expenses, {text, startDate, endDate, sortBy}) => {
     //    const startDateMatch = typeof startDate !== 'number' || expense.createdAt >= startDate; 
    //    const endDateMatch = typeof endDate !== 'number' || expense.createdAt <= endDate; 
    return expenses.filter((expense) => {
       const startDateMatch = startDate ? startDate.isSameOrBefore(moment(expense.createdAt), 'day') : true;
       const endDateMatch = endDate ? endDate.isSameOrAfter((expense.createdAt), 'day') :true;
       const textMatch = text == '' || expense.description.toLowerCase().includes(text.toLowerCase());

       return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if(sortBy === 'date') {
           return a.createdAt < b.createdAt ? 1 : -1;
        }
        else if(sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });
};

export default getVisibleExpenses;