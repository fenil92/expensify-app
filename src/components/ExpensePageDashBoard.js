import React from 'react';
import ExpensesList from './ExpensesList';
import ExpenseListFilter from './ExpenseListFilter';

const ExpensePageDashBoard = () => (
    <div>
        <ExpenseListFilter />
        <ExpensesList />
    </div>
);


export default ExpensePageDashBoard ;