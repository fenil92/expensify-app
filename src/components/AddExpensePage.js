import React from 'react';
import { connect } from 'react-redux';
import {startAddExpense} from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const AddExpensePage = (props) => (
    <div>
        <ExpenseForm  onSubmit={
            (expense) => {
               props.addExpense(expense);
               props.history.push('/');
            }
        }/>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    addExpense : (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);