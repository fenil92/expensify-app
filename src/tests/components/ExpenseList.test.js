import React from 'react';
import {ExpenseListComponent} from '../../components/ExpensesList';
import { shallow } from 'enzyme';
import {expenses} from '../fixtures/expenses';

test('should render expenselist correctly', () => {
    const wrapper = shallow(<ExpenseListComponent expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render expenselist correctly when there is no expense', () => {
    const wrapper = shallow(<ExpenseListComponent expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});