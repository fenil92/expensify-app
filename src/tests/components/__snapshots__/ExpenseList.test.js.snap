// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenselist correctly 1`] = `
<div>
  <ExpenseListItem
    amount={1200}
    createdAt={0}
    description="test"
    id={1}
    key="1"
    note="test note"
  />
  <ExpenseListItem
    amount={1500}
    createdAt={345600000}
    description="Coffee"
    id={2}
    key="2"
    note="Coffee note"
  />
  <ExpenseListItem
    amount={2200}
    createdAt={-432000000}
    description="Rent"
    id={3}
    key="3"
    note="rent note"
  />
  <ExpenseListItem
    amount={4200}
    createdAt={518400000}
    description="Gas"
    id={4}
    key="4"
    note="gas bill"
  />
</div>
`;

exports[`should render expenselist correctly when there is no expense 1`] = `
<div>
  <p>
    No Expenses
  </p>
</div>
`;
